[development] `make amaldebug` is failing (failing with `ld: library not found for -lduckdb_utf8proc`)

### What happens?

Running `make amaldebug` will result in `ld: library not found for -lduckdb_utf8proc`. Tried it on both macOS (arm) and Linux (amd64).

Happy for any pointers on how to get this working.

### To Reproduce

1. `git clone git@github.com:duckdb/duckdb.git`
2. `cd duckdb`
3. `make amaldebug`

It will end with:

```
[ 43%] Linking CXX executable ../../duckdb
ld: library not found for -lduckdb_utf8proc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [duckdb] Error 1
make[2]: *** [tools/shell/CMakeFiles/shell.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [debug] Error 2
```

### OS:

x64

### DuckDB Version:

Latest

### DuckDB Client:

C++

### Full Name:

Tom Nick

### Affiliation:

None

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

--- a/tools/sqlite3_api_wrapper/CMakeLists.txt
+++ b/tools/sqlite3_api_wrapper/CMakeLists.txt
@@ -13,7 +13,10 @@ include_directories(../../third_party/utf8proc/include)
 set(SQLITE_API_WRAPPER_FILES sqlite3_api_wrapper.cpp ${ALL_OBJECT_FILES})
 
 add_library(sqlite3_api_wrapper_static STATIC ${SQLITE_API_WRAPPER_FILES})
-target_link_libraries(sqlite3_api_wrapper_static duckdb_static duckdb_utf8proc)
+target_link_libraries(sqlite3_api_wrapper_static duckdb_static)
+if(NOT AMALGAMATION_BUILD)
+  target_link_libraries(sqlite3_api_wrapper_static duckdb_utf8proc)
+endif()
 link_threads(sqlite3_api_wrapper_static)
 
 if(NOT WIN32)

